src/Build/FileSystem/DirectoryCacheFileSystemWrapper.cs

```diff
@@ -87,8 +87,14 @@ private IEnumerable<string> EnumerateFullFileSystemPaths(string path, string sea
             {
                 return FileMatcher.IsAllFilesWildcard(searchPattern) || FileMatcher.IsMatch(fileName, searchPattern);
             };
-            FindTransform<string> transform = (ref ReadOnlySpan<char> fileName) => Path.Join(path.AsSpan(), fileName);
 
+#if !FEATURE_MSIOREDIST && NETFRAMEWORK && MONO
+            FindTransform<string> transform = (ref ReadOnlySpan<char> fileName) => path + '/' + fileName.ToString();
```

Comment:
This case is very confusing to me and it's surprising to see new code with an `#if MONO`. Can you elaborate on what it's needed for?

Do I understand correctly that there are three cases we care about now:

1. core (no `Microsoft.IO.Redist`)
2. VS/normal Framework (yes MIOR)
3. Framework-but-in-sourcebuild (no MIOR)

Should we call `Path.Combine(path, fileName.ToString())` in case 3?

Reply:
I did that because of the different path separators. I wouldn't expect it to matter, since I don't think we have a source build mono, but I included it for completeness.

Yes; those three cases. And yes, Path.Combine is better than splitting the MONO case from the !MONO case; Path.Join wasn't available, and I confused that momentarily with Path.Combine. I'll make that switch.